src/main/java/io/github/genomicdatainfrastructure/daam/services/AttachFileToApplicationService.java

```diff
@@ -20,17 +21,22 @@ public class AttachFileToApplicationService {
 
     private final String remsApiKey;
     private final RemsApplicationCommandApi remsApplicationCommandApi;
+    private final RemsApiQueryGateway gateway;
 
     @Inject
     public AttachFileToApplicationService(
             @ConfigProperty(name = "quarkus.rest-client.rems_yaml.api-key") String remsApiKey,
-            @RestClient RemsApplicationCommandApi remsApplicationCommandApi
+            @RestClient RemsApplicationCommandApi remsApplicationCommandApi,
+            RemsApiQueryGateway apiQueryGateway
     ) {
         this.remsApiKey = remsApiKey;
         this.remsApplicationCommandApi = remsApplicationCommandApi;
```

Comment:
Similarly, we can create a `RemsApiCommandGateway` and migrate all the code related to the `remsApplicationCommandApi`

Reply:
I wanted to do that, but I want first to validate the idea with you. If you agree, we can move on with this PR and tackle this idea afterwards. There are two PR's waiting to be create for the new endpoint, save application.

Anyway, the idea is to reorganize a bit the packages and introduce two interfaces, like you suggested before. However this requires a bit of refactoring.

Reply:
Yes no problem. This is not really related to your ticket anyway. Maybe we should create a special ticket to reorganize the project